haystack/core/pipeline/pipeline.py

```diff
@@ -78,6 +180,13 @@ def _run_component(self, name: str, inputs: Dict[str, Any]) -> Dict[str, Any]:
             res: Dict[str, Any] = instance.run(**inputs)
             self.graph.nodes[name]["visits"] += 1
 
+            # After a Component that has variadic inputs is run, we need to reset the variadic inputs that were consumed
+            for socket in instance.__haystack_input__._sockets_dict.values():  # type: ignore
+                if socket.name not in inputs:
+                    continue
+                if socket.is_variadic:
+                    inputs[socket.name] = []
```

Comment:
This was previously done from line 261 to 263. 
That was a mistake as that part of the logic was meant to only distribute the outputs of a Component run.

I decided to move it here as it makes more sense.